lib/ui/widgets/base_scaffold.dart

```diff
@@ -35,25 +39,69 @@ class BaseScaffold extends StatelessWidget {
         _isLoading = isLoading,
         _isInteractionLimitedWhileLoading = isInteractionLimitedWhileLoading;
 
+  @override
+  _BaseScaffoldState createState() => _BaseScaffoldState();
+}
+
+class _BaseScaffoldState extends State<BaseScaffold> {
+  // Displays text under the app bar when offline.
+  static bool _isOffline = false;
+
+  @override
+  void initState() {
+    super.initState();
+    Connectivity().onConnectivityChanged.listen((event) {
+      setState(() {
+        _isOffline = event == ConnectivityResult.none;
+      });
+    });
+  }
+
   @override
   Widget build(BuildContext context) => Scaffold(
-        appBar: appBar,
-        body: SafeArea(
-          top: false,
-          child: Stack(
-            children: [
-              body,
-              if (_isLoading)
-                buildLoading(
-                    isInteractionLimitedWhileLoading:
-                        _isInteractionLimitedWhileLoading)
-              else
-                const SizedBox()
-            ],
+        body: Scaffold(
+          appBar: widget.appBar,
+          body: SafeArea(
+            top: false,
+            child: Stack(
+              children: [
+                widget.body,
+                if (widget._isLoading)
+                  buildLoading(
+                      isInteractionLimitedWhileLoading:
+                          widget._isInteractionLimitedWhileLoading)
+                else
+                  const SizedBox()
+              ],
+            ),
           ),
+          bottomNavigationBar: widget._showBottomBar ? BottomBar() : null,
+          floatingActionButton: widget.fab,
+          floatingActionButtonLocation: widget.fabPosition,
         ),
-        bottomNavigationBar: _showBottomBar ? BottomBar() : null,
-        floatingActionButton: fab,
-        floatingActionButtonLocation: fabPosition,
+        bottomNavigationBar: _isOffline ? buildOfflineBar(context) : null,
       );
+
+  Widget buildOfflineBar(BuildContext context) {
+    return Stack(
+      alignment: Alignment.center,
+      children: [
+        Container(
+          color: _isLightTheme(context)
+              ? null
```

Comment:
just to be sure `null` put the background color right?

Reply:
It does yes!